gobblin-kafka/src/main/java/gobblin/kafka/schemareg/LiKafkaSchemaRegistry.java

```diff
@@ -0,0 +1,270 @@
+   *
+   * @param schema
+   * @return schema ID of the registered schema
+   * @throws SchemaRegistryException if registration failed
+   */
+  public synchronized MD5Digest register(Schema schema) throws SchemaRegistryException {
+    LOG.info("Registering schema " + schema.toString());
+
+    PostMethod post = new PostMethod(url);
+    post.addParameter("schema", schema.toString());
+
+    HttpClient httpClient = this.borrowClient();
+    try {
+      LOG.debug("Loading: " + post.getURI());
+      int statusCode = httpClient.executeMethod(post);
+      if (statusCode != HttpStatus.SC_CREATED) {
+        throw new SchemaRegistryException("Error occurred while trying to register schema: " + statusCode);
+      }
+
+      String response;
+      response = post.getResponseBodyAsString();
+      if (response != null) {
+        LOG.info("Received response " + response);
+      }
+
+      String schemaKey;
+      Header[] headers = post.getResponseHeaders(SCHEMA_ID_HEADER_NAME);
+      if (headers.length != 1) {
+        throw new SchemaRegistryException(
+            "Error reading schema id returned by registerSchema call: headers.length = " + headers.length);
+      } else if (!headers[0].getValue().startsWith(SCHEMA_ID_HEADER_PREFIX)) {
+        throw new SchemaRegistryException(
+            "Error parsing schema id returned by registerSchema call: header = " + headers[0].getValue());
+      } else {
+        LOG.info("Registered schema successfully");
+        schemaKey = headers[0].getValue().substring(SCHEMA_ID_HEADER_PREFIX.length());
+      }
+      MD5Digest schemaId = MD5Digest.fromString(schemaKey);
+      return schemaId;
+    } catch (Throwable t) {
+      throw new SchemaRegistryException(t);
+    } finally {
+      post.releaseConnection();
+      this.httpClientPool.returnObject(httpClient);
+    }
+  }
+
+  /**
+   * Fetch schema by key.
+   */
+  protected Schema fetchSchemaByKey(MD5Digest key) throws SchemaRegistryException {
+    String schemaUrl = this.url + GET_RESOURCE_BY_ID + key;
+
+    GetMethod get = new GetMethod(schemaUrl);
+
+    int statusCode;
+    String schemaString;
+    HttpClient httpClient = this.borrowClient();
+    try {
+      statusCode = httpClient.executeMethod(get);
+      schemaString = get.getResponseBodyAsString();
+    } catch (IOException e) {
+      throw new SchemaRegistryException(e);
+    } finally {
+      get.releaseConnection();
+      this.httpClientPool.returnObject(httpClient);
+    }
+
+    if (statusCode != HttpStatus.SC_OK) {
+      throw new SchemaRegistryException(
+          String.format("Schema with key %s cannot be retrieved, statusCode = %d", key, statusCode));
+    }
+
+    Schema schema;
+    try {
+      schema = new Schema.Parser().parse(schemaString);
+    } catch (Throwable t) {
+      throw new SchemaRegistryException(String.format("Schema with ID = %s cannot be parsed", key), t);
+    }
+
+    return schema;
+  }
+
+  public static void main(String[] args)
```

Comment:
This class is (mostly) a copy of the existing KafkaSchemaRegistry and KafkaAvroSchemaRegistry. 
I'm not sure why this code was there in the first place.